SP-2973: Updates for Q2 2026 (#63)
lsst/rtn-011: alertproduction.tex

During the Commissioning and Early Operations periods, alert packets for moving objects might not include the associated historical source records, and parameters such as phase curve slope (G) would be empty until sufficient detections exist to derive them.
Solar System discoveries will be reported to the Minor Planet Center on an ad-hoc basis during Early Operations, with a goal of increased automation as alert production becomes routine.

\subsubsection{Specifics of template generation in year 1}

In Year 1 (Y1), Rubin LSST will adopt an incremental and opportunistic approach to template generation. 

> Templates will be constructed as soon as sufficient high-quality visits have been accumulated in a given field and will be deployed for image differencing immediately upon validation. 

Reviewer: ```suggestion
Templates will be constructed once sufficient high-quality visits have been accumulated at a given sky position and will be deployed for image differencing upon validation. 
```

Let's drop "immediately" because the turnaround for incremental template production is not instant.

However, the production and scientific use of these templates must be understood within the broader operational realities of early survey execution.

The guiding principle for Y1 is that system calibration and survey integrity take precedence over alert production. 

> Following the guidance of the SCOC in Section 3.8 of \citeds{PSTN-056}, stating that releasing some alerts in Y1 is an important goal .. but that this objective must not override the priority of delivering a fully calibrated and stable system by the end of Y1.

Reviewer: ```suggestion
Following the guidance of the SCOC in Section 3.8 of \citeds{PSTN-056}, stating that releasing some alerts in Y1 is an important goal, but that this objective must not override the priority of delivering a fully calibrated and stable system by the end of Y1.
```

% \emph{releasing some alerts in Y1 is an important goal: it enables the time-domain and Solar System science communities to prepare for full-volume, full-fidelity alert streams in later years and increases early discovery potential. Early template generation is particularly useful for testing Solar System alert streams that require post-discovery re-detections. Nevertheless, this objective must not override the priority of delivering a fully calibrated and stable system by the end of Y1.}

> Templates will be generated once a field has accumulated:  A sufficient number of visits, of Adequate image qualityStable and validated calibration, Data that pass defined quality thresholds.

Reviewer: ```suggestion
Templates will be generated once a field has accumulated a sufficient number of visits of adequate image quality with stable and validated calibration and passing defined quality thresholds.
```

Once these conditions are met, templates will be built and activated without waiting for full annual data reprocessing. 
In principle, this allows templates to be used within the same observing season in which they are created.

> 

Reviewer: Once an incremental template is deployed, it will not be updated prior to DR1 to avoid repeated baseline shifts.

Author: I think this is stated in the text so no changes needed I don't think?

Reviewer: Ah, yes, I see it now.

In practice, however, incremental template generation is effectively a race between data acquisition, processing, and seasonal visibility. 
If sufficient visits are not accumulated and processed before a field sets, alerts for that field will not be issued until the next seasonal revisit. 
Given realistic Y1 conditions, this outcome is expected to be common.

> Operational Constraints in Year 1 inclkude Several factors  that limit the likelihood of widespread same-season template use indluding 

Reviewer: ```suggestion
Operational Constraints in Year 1 include several factors  that limit the likelihood of widespread same-season template use, including 
```

Alert generation in Y1 should therefore be viewed as opportunistic and demonstrative, not representative of steady-state LSST operations. Limited regions may produce alerts once templates are available, potentially resulting in early bursts of alerts in specific areas. As coverage evolves, alert rates may fluctuate significantly.

The objective of Y1 alert release is to:
\begin{itemize}

\item Exercise the end-to-end alert system under realistic conditions,

\item Enable community preparation and pipeline testing,

\item Demonstrate operational capability ahead of full survey scale.

\end{itemize}
Full-volume, uniform alert streams remain a goal for subsequent years.

In summary, LSST will actively pursue incremental template generation during Year 1 and will deploy templates as soon as they meet defined quality standards. However, realistic operational constraints make widespread same-season template use unlikely. Alert production in Y1 will be limited and opportunistic, with calibration stability and survey quality remaining the overriding priorities.

\subsubsection{Using external templates}

> Some new External Drivers and Pathfinders including  The upcoming six-month LVK run (“IR1”) provides motivation to deploy DECam-based templates into production. 

Reviewer: ```suggestion
Some new external drivers including the upcoming six-month LVK run (“IR1”) provides motivation to consider deploying DECam-based templates into in Alert Production. 
```

> These will allow the alert system to build operational history, validate performance, and demonstrate readiness ahead of full LSST alert volumes. Such efforts complement, but do not replace, LSST template generation in Y1.

Reviewer: ```suggestion
These would allow the alert system to build operational history, validate performance, and demonstrate readiness ahead of full LSST alert volumes.  This approach is under investigation; deployment depends on demonstrating an acceptably low rate of false positives.
```

(Removing the "complement but not replace"--if DECam subtractions are successful we would not hurry to build Rubin templates incrementally)

They will be distributed only as per-image Butler datasets and will not be accessible via database queries.

> Prompt image publication is under active development.

Reviewer 2: ```suggestion
Prompt image publication is under active development, and is a particular challenge given the observatory's goal of delivering Data Preview 2, prompt image and catalog products, and access to the PPDB all within the same 6 month period.  
```

Author: it is a good idea to make it clear that the challenges, and delays for the PPDB and NPC are purely technical in nature

Author: as the document has since been updated, I'll add this in manually. Thanks @drphilmarshall

> Nightly PVIs and per-epoch direct-image \texttt{Source} catalogs are  not expected to begin before July  2026.

Reviewer 2: "Not before X" is a semi-infinite date range that does not capture our actual uncertainty on the NPC milestone. I have heard @ebellm estimate "before the end of the calendar year", for example: is December 15 not a reasonable late date for the NPC milestone? Is it the right late date for the PPDB milestone too?

Author: @ebellm  - this is for you

Reviewer: Not actually for me in an org chart sense: data publication is not mine.  I don't want to drop dates in that I personally don't direct the work for.

Reviewer 3: So while I take Phil's point that "not before" is unbound in math space, I think this usage here is perfectly appropriate.
* I think prompt and accurate communication is better in all cases. "Not before X" communicates the bit we actually know with high confidence - it's not going to be before X. If we could give a reliable date range we would have done that, and making up a "well definitely before 2035" type of date range for the sake of having a date range doesn't help anybody. 
* There is an evident widely understood colloquial use of "not before" that nobody would think it means never. If my plumber says "I can't fit you in before May" I understand perfectly well that I am not getting the work done beore May and he'll get back to me sometime after May that is consistent with him doing my plumbing. If he wanted to say "I can't do your plumbing on any timescale that you would find acceptable" he would just say that...

I would rather get the "not before" statement out so users can plan their lives rather than give a poor fidelity estimate.

Reviewer 2: Circling back to this after a couple of very interesting discussions with @leanne on zoom and @frossie on Slack. I think the issue with "not before July" in the current context is that our readers don't have the same good intuition about Rubin's various milestones that they do about plumbers, and so they might draw an overly pessimistic conclusion when given such an indication - especially since we've tried to be consistent about communicating uncertainty via approximate probability distributions in the form of date ranges. So, we had some back and forth about the complex challenge of shipping DP2, the PPDB and the nightly PVIs and catalogs all in the same 6 month period(!), and my conclusion is that there's a good chance of it all being done by the end of September. I'll suggest changes to match this, and also try and find a way to flag in the text the particular difficulty of estimating the prompt product schedule, so that if (at ~10% probabilty) we do have to extend this range folks won't be too surprised. For anyone following this from outside Rubin, I'll just add that the Data Management team are moving mountains to get data out to the science community - the LSST Big Data problems start at home before being delivered to the rest of you!

> Nightly PVIs and per-epoch direct-image \texttt{Source} catalogs are  not expected to begin before July  2026.

Reviewer 2: ```suggestion
At present, nightly PVIs and per-epoch direct-image \texttt{Source} catalogs are expected to be made available between June and September 2026.
```
